security/providers/http-auth/src/main/java/io/helidon/security/providers/httpauth/HttpBasicAuthProvider.java

```diff
@@ -116,54 +128,67 @@ public boolean isOutboundSupported(ProviderRequest providerRequest,
 
     @Override
     protected OutboundSecurityResponse syncOutbound(ProviderRequest providerRequest,
                                                     SecurityEnvironment outboundEnv,
                                                     EndpointConfig outboundEp) {
 
-        // first resolve user to use
+        // explicit username in request properties
         Optional<Object> maybeUsername = outboundEp.abacAttribute(EP_PROPERTY_OUTBOUND_USER);
         if (maybeUsername.isPresent()) {
             String username = maybeUsername.get().toString();
             char[] password = passwordFromEndpoint(outboundEp);
 
-            return toBasicAuthOutbound(username, password);
+            return toBasicAuthOutbound(outboundEnv,
+                                       HttpBasicOutboundConfig.DEFAULT_TOKEN_HANDLER,
+                                       username,
+                                       password);
         }
 
-        // and if not present, use the one from request
-        SecurityContext secContext = providerRequest.securityContext();
-
-        // first try user
-        Optional<BasicPrivateCredentials> creds = secContext.user()
-                .flatMap(this::credentialsFromSubject);
+        var target = outboundConfig.findTargetCustomObject(outboundEnv,
+                                                           HttpBasicOutboundConfig.class,
+                                                           HttpBasicOutboundConfig::create,
+                                                           HttpBasicOutboundConfig::create);
 
-        if (!creds.isPresent()) {
-            // if not present, try service
-            creds = secContext.service()
-                    .flatMap(this::credentialsFromSubject);
+        if (target.isEmpty()) {
+            return OutboundSecurityResponse.abstain();
         }
 
-        Optional<char[]> overridePassword = outboundEp.abacAttribute(EP_PROPERTY_OUTBOUND_PASSWORD)
-                .map(String::valueOf)
-                .map(String::toCharArray);
+        HttpBasicOutboundConfig outboundConfig = target.get();
+
+        if (outboundConfig.hasExplicitUser()) {
+            // use configured user
+            return toBasicAuthOutbound(outboundEnv,
+                                       outboundConfig.tokenHandler(),
+                                       outboundConfig.explicitUser(),
+                                       outboundConfig.explicitPassword());
+        } else {
+            // propagate current user (if possible)
+            SecurityContext secContext = providerRequest.securityContext();
+            // first try user
+            Optional<BasicPrivateCredentials> creds = secContext.user()
+                    .flatMap(this::credentialsFromSubject);
+            if (creds.isEmpty()) {
+                // if not present, try service
+                creds = secContext.service()
+                        .flatMap(this::credentialsFromSubject);
+            }
```

Comment:
This could also work
```java
Optional<BasicPrivateCredentials> creds = secContext.user()
                    .flatMap(this::credentialsFromSubject)
                    .or(secContext::service)
                    .flatMap(this::credentialsFromSubject);
```

Reply:
Yes, but this is as it was and I do not want to make further changes unless it is an actual problem.